Remotion/Data/DomainObjects.UberProfIntegration/EntityFramworkExtension.cs

```diff
@@ -0,0 +1,269 @@
+    {
+    }
+
+    public QueryResult<T> FilterQueryResult<T> (ClientTransaction clientTransaction, QueryResult<T> queryResult) where T : DomainObject
+    {
+      return queryResult;
+    }
+
+    public void Committing (ClientTransaction clientTransaction, IReadOnlyList<DomainObject> domainObjects, ICommittingEventRegistrar eventRegistrar)
+    {
+    }
+
+    public void CommitValidate (ClientTransaction clientTransaction, IReadOnlyList<PersistableData> committedData)
+    {
+    }
+
+    public void Committed (ClientTransaction clientTransaction, IReadOnlyList<DomainObject> domainObjects)
+    {
+    }
+
+    public void RollingBack (ClientTransaction clientTransaction, IReadOnlyList<DomainObject> domainObjects)
+    {
+    }
+
+    public void RolledBack (ClientTransaction clientTransaction, IReadOnlyList<DomainObject> domainObjects)
+    {
+    }
+
+
+    #endregion
+
+    #region Implementation of IPersistenceExtension
+
+    public void ConnectionOpened (Guid connectionID)
+    {
+    }
+
+    public void ConnectionClosed (Guid connectionID)
+    {
+    }
+
+    #endregion
+
+    private readonly EntityFrameworkAppenderProxy _appenderProxy;
+    private readonly Guid _clientTransactionID;
+
+    public EntityFrameworkExtension (Guid clientTransactionID, EntityFrameworkAppenderProxy appenderProxy)
+    {
+      ArgumentUtility.CheckNotNull("appenderProxy", appenderProxy);
+
+      _clientTransactionID = clientTransactionID;
+      _appenderProxy = appenderProxy;
+    }
+
+    public EntityFrameworkAppenderProxy AppenderProxy
+    {
+      get { return _appenderProxy; }
+    }
+
+    public Guid ClientTransactionID
+    {
+      get { return _clientTransactionID; }
+    }
+
+    public string Key
+    {
+      get { return typeof(EntityFrameworkExtension).GetFullNameChecked(); }
+    }
+
+    public void TransactionInitialize (ClientTransaction clientTransaction)
+    {
+      _appenderProxy.ConnectionStarted(_clientTransactionID);
+    }
+
+    public void TransactionDiscard (ClientTransaction clientTransaction)
+    {
+      _appenderProxy.ConnectionDisposed(_clientTransactionID);
+    }
+
+    public void QueryCompleted (Guid connectionID, Guid queryID, TimeSpan durationOfDataRead, int rowCount)
+    {
+      _appenderProxy.StatementRowCount(_clientTransactionID, queryID, rowCount);
+    }
+
+    public void QueryError (Guid connectionID, Guid queryID, Exception e)
+    {
+      _appenderProxy.StatementError(_clientTransactionID, e);
+    }
+
+    public void QueryExecuted (Guid connectionID, Guid queryID, TimeSpan durationOfQueryExecution)
+    {
+      _appenderProxy.CommandDurationAndRowCount(_clientTransactionID, durationOfQueryExecution.Milliseconds, null);
+    }
+
+    public void QueryExecuting (
+        Guid connectionID, Guid queryID, string commandText, IDictionary<string, object?> parameters)
```

Comment:
```suggestion
    public void QueryExecuting (Guid connectionID, Guid queryID, string commandText, IDictionary<string, object?> parameters)
```
should be okay in current styleguide